client/asset/dcr/wallet.go

```diff
@@ -92,54 +88,65 @@ type Wallet interface {
+	// returned block header is a wire.BlockHeader with the addition of the block's
+	// median time.
+	GetBlockHeader(ctx context.Context, blockHash *chainhash.Hash) (*BlockHeader, error)
+	// GetBlock returns the *wire.MsgBlock.
+	GetBlock(ctx context.Context, blockHash *chainhash.Hash) (*wire.MsgBlock, error)
 	// GetTransaction returns the details of a wallet tx, if the wallet contains a
 	// tx with the provided hash. Returns asset.CoinNotFoundError if the tx is not
 	// found in the wallet.
 	GetTransaction(ctx context.Context, txHash *chainhash.Hash) (*walletjson.GetTransactionResult, error)
-	// GetRawTransactionVerbose returns details of the tx with the provided hash.
+	// GetRawTransaction returns details of the tx with the provided hash.
 	// Returns asset.CoinNotFoundError if the tx is not found.
-	GetRawTransactionVerbose(ctx context.Context, txHash *chainhash.Hash) (*chainjson.TxRawResult, error)
-	// GetRawMempool returns hashes for all txs of the specified type in the node's
-	// mempool.
-	GetRawMempool(ctx context.Context, txType chainjson.GetRawMempoolTxTypeCmd) ([]*chainhash.Hash, error)
+	GetRawTransaction(ctx context.Context, txHash *chainhash.Hash) (*wire.MsgTx, error)
 	// GetBestBlock returns the hash and height of the wallet's best block.
 	GetBestBlock(ctx context.Context) (*chainhash.Hash, int64, error)
 	// GetBlockHash returns the hash of the mainchain block at the specified height.
 	GetBlockHash(ctx context.Context, blockHeight int64) (*chainhash.Hash, error)
-	// BlockCFilter fetches the block filter info for the specified block.
-	BlockCFilter(ctx context.Context, blockHash *chainhash.Hash) (filter, key string, err error)
-	// LockWallet locks the wallet.
-	LockWallet(ctx context.Context) error
-	// UnlockWallet unlocks the wallet.
-	UnlockWallet(ctx context.Context, passphrase string, timeoutSecs int64) error
-	// WalletUnlocked returns true if the wallet is unlocked.
-	WalletUnlocked(ctx context.Context) bool
-	// AccountUnlocked returns true if the specified account is unlocked.
-	AccountUnlocked(ctx context.Context, account string) (*walletjson.AccountUnlockedResult, error)
-	// LockAccount locks the specified account.
-	LockAccount(ctx context.Context, account string) error
-	// UnlockAccount unlocks the specified account.
-	UnlockAccount(ctx context.Context, account, passphrase string) error
+	// MatchAnyScript looks for any of the provided scripts in the block specified.
+	MatchAnyScript(ctx context.Context, blockHash *chainhash.Hash, scripts [][]byte) (bool, error)
+	// AccountUnlocked returns true if the account is unlocked.
+	AccountUnlocked(ctx context.Context, acctName string) (bool, error)
+	// LockAccount locks the account.
+	LockAccount(ctx context.Context, acctName string) error
+	// UnlockAccount unlocks the Wallet. ExchangeWallet does not differentiate
+	// account locking vs wallet locking, but the underlying implementation may
+	// choose to unlock only the account.
```

Comment:
Going through this again, the doc reads somehow. `UnlockAccount unlocks the Wallet`. Should that be `UnlockWallet`? I also notice that the `Un/lockWallet` methods are removed, so implementations have to understand that this single method may do either wallet unlocking or single account unlocking, whichever is necessary to provide access to the account priv keys. I think this should go in the doc for this and related methods.

Reply:
I would even suggest just `Unlock`, but as I've said before, I would avoid that at all costs because it is ungrepable.  @itswisdomagain is more deep in multi-account support, so whatever he thinks I guess.  Or you could adjust this later as you see fit in your PR Wisdom.